Makefile

```diff
@@ -60,17 +60,37 @@ PROJECT_CMODULES   += crc32 compress deflate trees
 
 PROJECT_OCAML_LIBS = unix str #OCaml changed libstr into libcamlstr and "you are not supposed to link with -lstr"
 PROJECT_LIBS       = unix camlstr
+
+PROJECT_RESOURCES = weidu_resources
+
 .PHONY: weidu
 weidu:  $(PROJECT_EXECUTABLE)
 $(PROJECT_EXECUTABLE) : $(PROJECT_MODULES:%=$(OBJDIR)/%.$(CMO)) \
                         $(PROJECT_CMODULES:%=$(OBJDIR)/%.$(OBJEXT))
 	@$(NARRATIVE) Linking $(COMPILETOWHAT) $@
+	
+ifeq (,$(filter $(uname),"Linux" "Darwin"))
 	$(CAMLLINK) -o $@ \
-                    $(PROJECT_OCAML_LIBS:%=%.$(CMXA)) \
-                    $(PROJECT_LIBS:%=-cclib -l%) \
-                    $(PROJECT_CLIBS:%=-cclib %) \
-                    $^
-	cp $(PROJECT_EXECUTABLE) .
+		$(PROJECT_OCAML_LIBS:%=%.$(CMXA)) \
+		$(PROJECT_LIBS:%=-cclib -l%) \
+		$(PROJECT_CLIBS:%=-cclib %) \
+		$^
+		cp $(PROJECT_EXECUTABLE) .
+else
+	cat windows_resources/weidu_resources.rc | sed -e "s/FILEVERSION\W\+[[:digit:]]\+,[[:digit:]]\+,[[:digit:]]\+,[[:digit:]]\+/FILEVERSION\t\t$(grep 'version = ' src/version.ml | cut -d'"' -f2 -z | cut -c-3),$(grep 'version = ' src/version.ml | cut -d'"' -f2 -z | cut -c4-),0,0/gI" > windows_resources/weidu_resources.rc
```

Comment:
Not only is cat not needed here, but this will randomly cause problems, since you're reading and writing to the same file at the same time. Either use a temporary file or just sed -i, which does in-place substitution.
But this monstrous parsing can also be simplified.

Reply:
Removed cat
use sed -i
parsing could be improved but I'm not bash/regex expert, right now it's very simple, easy to understand and works